src/app/content/components/Page/highlightManager.ts

```diff
@@ -49,8 +55,12 @@ const onSelectHighlight = (
   services: Services,
   highlights: Highlight[],
   highlight: Highlight | undefined
-) => defer(() => {
-  if (highlights.length > 0 || !highlight || services.getProp().focused) {
+) => defer(async() => {
+  if (highlights.length > 0 || !highlight) {
+    return;
+  }
+
+  if (services.getProp().focused && services.getProp().hasUnsavedHighlight && !await showConfirmation()) {
     return;
```

Comment:
I thought that it might be a good idea to add `assertWindow().getSelection().removeAllRangers()` here but I'm having trouble testing it. I'm assuming it's because of 
`  afterEach(() => {
    delete window.document.getSelection;
  });`?

Reply:
thats a good idea, go ahead and add it. i can help with testing if necessary. i think the selection object is mocked so it might be a case of needing to add that `removeAllRangers` to the mock in order to test that it is called (and not have the code fail when it is not there)